### PR TITLE
error logging fixes

### PR DESCRIPTION
**Pull request by Orzelius**
please see commits

related to https://github.com/garden-io/garden/issues/4687

### CODE REVIEW

File: core/src/plugins/exec/build.ts [hunk 1]
```diff
@@ -70,11 +72,11 @@ export const execBuildHandler = execBuild.addHandler("build", async ({ action, l
       renderMessageWithDivider({
         prefix,
         msg: output.detail?.buildLog,
-        isError: false,
+        isError: success,
```
Comment (vvagaytsev) [inline core/src/plugins/exec/build.ts:75]: Should this be negated? `isError: !success`

File: core/src/plugins/exec/deploy.ts [hunk 2]
```diff
@@ -193,22 +194,24 @@ execDeploy.addHandler("deploy", async (params) => {
...
-    if (outputLog) {
+    if (result.outputLog) {
       const prefix = `Finished deploying service ${chalk.white(action.name)}. Here is the output:`
       log.verbose(
         renderMessageWithDivider({
           prefix,
-          msg: outputLog,
-          isError: false,
+          msg: result.outputLog,
+          isError: result.success,
```
Comment (vvagaytsev) [inline core/src/plugins/exec/deploy.ts:203]: [Same](https://github.com/garden-io/garden/pull/4746/commits/4f30027ddb3b972523b45942862b80ca1580cbcd#r1252165419) here.

File: core/src/plugins/exec/run.ts [hunk 3]
```diff
@@ -54,7 +54,7 @@ execRun.addHandler("run", async ({ artifactsPath, log, action, ctx }) => {
       renderMessageWithDivider({
         prefix,
         msg: outputLog,
-        isError: false,
+        isError: success,
```
Comment (vvagaytsev) [inline core/src/plugins/exec/run.ts:57]: [Same](https://github.com/garden-io/garden/pull/4746/commits/4f30027ddb3b972523b45942862b80ca1580cbcd#r1252165419) here.

File: core/src/plugins/exec/test.ts [hunk 4]
```diff
@@ -42,14 +42,13 @@ execTest.addHandler("run", async ({ log, action, artifactsPath, ctx }) => {
...
-  if (outputLog) {
+  if (result.outputLog) {
     const prefix = `Finished executing ${chalk.white(action.key())}. Here is the full output:`
     log.verbose(
       renderMessageWithDivider({
         prefix,
-        msg: outputLog,
-        isError: false,
+        msg: result.outputLog,
+        isError: result.success,
```
Comment (vvagaytsev) [inline core/src/plugins/exec/test.ts:51]: [Same](https://github.com/garden-io/garden/pull/4746/commits/4f30027ddb3b972523b45942862b80ca1580cbcd#r1252165419) here.

File: core/src/plugins/exec/build.ts (hunk 1 above)
Comment (Orzelius) [reply core/src/plugins/exec/build.ts]: Fixed in [3ded16d](https://github.com/garden-io/garden/pull/4746/commits/3ded16dff1a2aeb3c3efdca8a4806d537e7d1be9) but it has no effect as the `isError` property is used only for color which is overridden anyway.

**Comment by Orzelius** [comment]
All test failures seem to be flakes. Please re-review.

**Review by vvagaytsev — APPROVED** [review]
Thanks! 👍

**Pull request merged**